graphql_compiler/macros/macro_edge/validation.py

```diff
@@ -78,24 +74,12 @@ def _validate_macro_ast_directives(ast, inside_fold_scope=False):
         inside_fold_scope: bool, whether the subtree is within a @fold scope
     """
     subselection_inside_fold_scope = inside_fold_scope
-    directives_with_no_restrictions = frozenset({
-        FilterDirective.name,
-        TagDirective.name,
-        OptionalDirective.name,
-        RecurseDirective.name,
-        MacroEdgeDefinitionDirective.name,
-    })
-    disallowed_directives = frozenset({
-        OutputDirective.name,
-        OutputSourceDirective.name,
-    })
     for directive in ast.directives:
         name = directive.name.value
-        if name in directives_with_no_restrictions:
+        if name in DIRECTIVES_ALLOWED_IN_MACRO_EDGE_DEFINITION:
+            pass
+        elif name == MacroEdgeDefinitionDirective.name:
```

Comment:
Shouldn't the MacroEdgeDefinition directive be in the DIRECTIVES_ALLOWED_IN_MACRO_EDGE_DEFINTION set?

Reply:
Renamed .._ALLOWED_... to .._OPTIONAL_.. to better indicate the directives listed are not related to macros.